native/arguments.go

```diff
@@ -184,25 +187,38 @@ func (j JarArguments) Configure(inputArgs []string) ([]string, string, error) {
 	startClass := strings.TrimSuffix(jarFileName, ".jar")
 
 	if containsArg("-jar", inputArgs) {
-		var tmpArgs []string
-		var skip bool
-		for _, inputArg := range inputArgs {
-			if skip {
-				skip = false
-				break
-			}
-
-			if inputArg == "-jar" {
-				skip = true
-				break
-			}
-
-			tmpArgs = append(tmpArgs, inputArg)
-		}
-		inputArgs = tmpArgs
+		inputArgs = replaceJarArguments(inputArgs)
 	}
-
 	inputArgs = append(inputArgs, "-jar", candidates[0])
 
 	return inputArgs, startClass, nil
 }
+
+func replaceJarArguments(fileArgs []string) []string{
```

Comment:
Do you feel like there is enough test coverage of this method? I think we are testing it though the `jar file` context in arguments_test.go, but we don't really cover it through the `user arguments from file` context. They both end up calling this method, but it might be nice to have one test in `user arguments from file` that actually writes to that file & triggers the replace method, so we can validate that code which sets up and calls this method is correct.

Reply:
Suggestion:

In Before()...

```
Expect(ioutil.WriteFile(filepath.Join(ctx.Application.Path, "target", "more-stuff-quotes.txt"), []byte(`before -jar "more stuff.jar" after`), 0644)).To(Succeed())
```

and test...

```
it("works with quotes in the file", func() {
			inputArgs := []string{"one", "two", "three"}
			args, startClass, err := native.UserFileArguments{
				ArgumentsFile: filepath.Join(ctx.Application.Path, "target/more-stuff-quotes.txt"),
			}.Configure(inputArgs)
			Expect(err).ToNot(HaveOccurred())
			Expect(startClass).To(Equal(""))
			Expect(args).To(HaveLen(4))
			Expect(args).To(Equal([]string{"one", "two", "three", fmt.Sprintf("@%s", filepath.Join(ctx.Application.Path, "target/more-stuff-quotes.txt"))}))
			bits, err := ioutil.ReadFile(filepath.Join(ctx.Application.Path, "target/more-stuff-quotes.txt"))
			Expect(err).ToNot(HaveOccurred())
			Expect(string(bits)).To(Equal("before after"))
		})
```

This should strip out the `-jar <file>` part, but in this case I put in a space so it's broken :( 

```
FAIL | Unit (0.02s)
FAIL |   Unit/native (0.02s)
FAIL |         Unit/native/Arguments/user_arguments_from_file/works_with_quotes_in_the_file (0.00s)
     |             arguments_test.go:162:
     |                 Expected
     |                     <string>: before stuff.jar\" after
     |                 to equal
     |                     <string>: before after
FAIL | 	github.com/paketo-buildpacks/native-image/v5/native	0.142s
```

Reply:
This worked for me to make it support spaces. Not saying it's perfect, but it makes the test above pass:

```
func replaceJarArguments(fileArgs []string) []string {
	var tmpArgs, modifiedArgs []string
	var skip, skipTillQuote bool
	var className string

	for i, inputArg := range fileArgs {
		if strings.HasPrefix(inputArg, `"`) {
			skipTillQuote = true
			continue
		}

		if skipTillQuote && strings.HasSuffix(inputArg, `"`) {
			skipTillQuote = false
		}

		if skip {
			skip = false
			className = strings.TrimSuffix(fileArgs[i], ".jar")
			continue
		}

		if inputArg == "-jar" {
			skip = true
			continue
		}

		tmpArgs = append(tmpArgs, inputArg)
	}

	for _, inputArg := range tmpArgs {
		if inputArg == className {
			continue
		}
		modifiedArgs = append(modifiedArgs, inputArg)
	}
	return modifiedArgs
}
```

Reply:
Added the test and func improvment. Also added a test to confirm we remove the class name if it is found since this func also performs this functionality.